### ISSUE TITLE
[Workshop] Introduction To Unit Testing in JavaScript

### ISSUE THREAD

**Issue by ChrisOwen101** [CodeYourFuture/syllabus#33]
# Workshop Request

## What is the title of your workshop

Introduction To Unit Testing in JavaScript

## What are the key topics that will be covered in the workshop?

By the end of this session and homework, a student should be able to 

- Define what Unit Testing is
- Give reasons for why testing - and unit testing in particular - is important
- Read testing code and describe what it is testing 
- Run tests on a JavaScript codebase and tell if they have passed or failed
- Write a unit test that tests a Javascript function
- Refactor code to make it testable and then write a test for it

## What knowledge does the student need before starting?

The students will 

- have knowledge of JavaScript 
- be able to write JavaScript code to an average level
   - they will have completed up to [JS3](https://syllabus.codeyourfuture.io/js-core-3/index) of our syllabus
- have no knowledge of what testing is or why it is useful

## What are the topics that will **not** be covered in the workshop?

- Any other forms of testing (e.g. UI testing)
   - Except in passing to describe the difference from Unit Testing
- 

## Any other notes?

This workshop must also include homework for the students to study in the time after the lesson. It is expected that a lot of the learning will happen in the homework section.

This workshop could be based on [this lesson](https://syllabus.codeyourfuture.io/workshops/js-testing-workshop) which was recently cut from our syllabus because of time constraints in the course.

This workshop could also be split across two or more lessons if the content is deemed to be to lengthy.

This workshop must include practical exercises for the students to try during the lesson.

## Completion

When completed the lesson plan, slides and exercises should be added to the Workshop section of the Syllabus
